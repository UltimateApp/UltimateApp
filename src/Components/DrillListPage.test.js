import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Store/testStore';
import { createDrill } from '../Fixtures/TestFixtures';
import { DrillTypes } from '../Fixtures';

import ConnectedDrillListPage, { DrillListPage } from './DrillListPage';

describe('<DrillListPage />', () => {
  const route = {
    params: {
      type: DrillTypes.TECHNICAL,
    },
  };

  it('renders correctly when connected', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedDrillListPage route={route} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a drill', () => {
    const drills = [createDrill()];
    const tree = renderer.create(<DrillListPage route={route} drills={drills} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
