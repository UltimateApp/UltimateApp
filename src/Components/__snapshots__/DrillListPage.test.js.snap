// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillListPage /> renders correctly when connected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingLeft": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A6A6A6",
        "marginBottom": 20,
      }
    }
  >
    2
     drills available
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "animation": Object {
            "ids": Array [
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "offense",
              "offense",
              "offense",
              "offense",
              "offense",
              "disc",
            ],
            "positions": Array [
              Array [
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.2,
                    0.45,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.45,
                    0.7,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.7,
                    0.45,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
              ],
              Array [
                Array [
                  Array [
                    0.16,
                    0.3,
                  ],
                ],
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                Array [
                  Array [
                    0.7,
                    0.45,
                  ],
                ],
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                Array [
                  Array [
                    0.45,
                    0.7,
                  ],
                ],
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                Array [
                  Array [
                    0.2,
                    0.45,
                  ],
                ],
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
              ],
              Array [
                Array [
                  Array [
                    0.36,
                    0.36,
                  ],
                ],
                undefined,
                Array [
                  Array [
                    0.58,
                    0.36,
                  ],
                ],
                Array [
                  Array [
                    0.58,
                    0.58,
                  ],
                ],
                Array [
                  Array [
                    0.36,
                    0.58,
                  ],
                ],
                Array [
                  Array [
                    0.36,
                    0.36,
                  ],
                ],
              ],
            ],
            "texts": Array [
              "",
              "",
              "",
              "",
              "",
              "",
              "",
              "",
              "1",
              "2",
              "3",
              "4",
              "5",
              "",
            ],
          },
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget",
          "durationInMinutes": 30,
          "equipment": "Equipment needs for the drill",
          "equipmentLabel": "",
          "goals": Array [
            "Defense",
          ],
          "id": 3,
          "image": "https://i1.wp.com/www.newdelhitimes.com/wp-content/uploads/2019/07/AP19204604544330.jpg?w=1024&ssl=1",
          "intensity": "Low",
          "level": "Beginner",
          "minimalPlayersNumber": 2,
          "seasonTiming": "",
          "source": "Author",
          "steps": Array [],
          "title": "Frisbee Drill Title",
          "type": "technical",
        },
        Object {
          "animation": Object {
            "ids": Array [
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "triangle",
              "offense",
              "offense",
              "offense",
              "offense",
              "offense",
              "disc",
            ],
            "positions": Array [
              Array [
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.2,
                    0.45,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.45,
                    0.7,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.7,
                    0.45,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
              ],
              Array [
                Array [
                  Array [
                    0.16,
                    0.3,
                  ],
                ],
                Array [
                  Array [
                    0.45,
                    0.2,
                  ],
                ],
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.3,
                  ],
                ],
                undefined,
                Array [
                  Array [
                    0.7,
                    0.45,
                  ],
                ],
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.6,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                Array [
                  Array [
                    0.45,
                    0.7,
                  ],
                ],
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
              ],
              Array [
                Array [
                  Array [
                    0.3,
                    0.6,
                  ],
                ],
                undefined,
                undefined,
                undefined,
                Array [
                  Array [
                    0.2,
                    0.45,
                  ],
                ],
                Array [
                  Array [
                    0.3,
                    0.3,
                  ],
                ],
              ],
              Array [
                Array [
                  Array [
                    0.36,
                    0.36,
                  ],
                ],
                undefined,
                Array [
                  Array [
                    0.58,
                    0.36,
                  ],
                ],
                Array [
                  Array [
                    0.58,
                    0.58,
                  ],
                ],
                Array [
                  Array [
                    0.36,
                    0.58,
                  ],
                ],
                Array [
                  Array [
                    0.36,
                    0.36,
                  ],
                ],
              ],
            ],
            "texts": Array [
              "",
              "",
              "",
              "",
              "",
              "",
              "",
              "",
              "1",
              "2",
              "3",
              "4",
              "5",
              "",
            ],
          },
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget",
          "durationInMinutes": 30,
          "equipment": "Equipment needs for the drill",
          "equipmentLabel": "",
          "goals": Array [
            "Defense",
          ],
          "id": 4,
          "image": "https://www.plu.edu/recreations/wp-content/uploads/sites/197/2019/04/ultimate-frisbee-3-2-19-1122-scaled-1536x1163.jpg",
          "intensity": "Low",
          "level": "Beginner",
          "minimalPlayersNumber": 2,
          "seasonTiming": "",
          "source": "Author",
          "steps": Array [],
          "title": "Frisbee Drill Title",
          "type": "technical",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": 80,
              "marginBottom": 20,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://i1.wp.com/www.newdelhitimes.com/wp-content/uploads/2019/07/AP19204604544330.jpg?w=1024&ssl=1",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 80,
                "marginRight": 10,
                "width": 80,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 5,
                "paddingBottom": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Author
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Frisbee Drill Title
            </Text>
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Duration: 
              30
               min - players: 
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": 80,
              "marginBottom": 20,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.plu.edu/recreations/wp-content/uploads/sites/197/2019/04/ultimate-frisbee-3-2-19-1122-scaled-1536x1163.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 80,
                "marginRight": 10,
                "width": 80,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 5,
                "paddingBottom": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Author
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Frisbee Drill Title
            </Text>
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Duration: 
              30
               min - players: 
              2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 50,
        "bottom": "5%",
        "elevation": 10,
        "height": 100,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": "5%",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
        "width": 100,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<DrillListPage /> renders correctly with a drill 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingLeft": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A6A6A6",
        "marginBottom": 20,
      }
    }
  >
    0
     drills available
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 50,
        "bottom": "5%",
        "elevation": 10,
        "height": 100,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": "5%",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
        "width": 100,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
